daemon/daemon_main.go

```diff
@@ -1109,6 +1098,50 @@ func runDaemon() {
 	bootstrapStats.k8sInit.Start()
 	d.initK8sSubsystem()
 	bootstrapStats.k8sInit.End(true)
+	// If K8s is enabled we can do the service translation automagically by
+	// looking at services from k8s and retrieve the service IP from that.
+	// This makes cilium to not depend on kube dns to interact with etcd operator
+	var goopts *kvstore.ExtraOptions
+	if k8s.IsEnabled() {
+		goopts = &kvstore.ExtraOptions{}
+		goopts.DialOption = []grpc.DialOption{
+			grpc.WithDialer(func(s string, duration time.Duration) (conn net.Conn, e error) {
+				// typical service name "cilium-etcd-client.kube-system.svc"
+				idx1 := strings.IndexByte(s, '.')
+				if idx1 >= 0 {
```

Comment:
Move this into a function so you can write a quick unit test for the string parsing logic